examples/cohort_replicator_kafka_pg/examples/cohort_replicator_kafka_pg.rs

```diff
@@ -0,0 +1,100 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use cohort_banking::{
+    callbacks::statemap_installer::StatemapInstallerImpl,
+    state::postgres::{database::Database, database_config::DatabaseConfig},
+};
+use talos_certifier::{env_var, env_var_with_defaults, ports::MessageReciever};
+use talos_certifier_adapters::{KafkaConfig, KafkaConsumer};
+use talos_cohort_replicator::{talos_cohort_replicator, CohortReplicatorConfig, ReplicatorSnapshot};
+
+use cohort_banking::state::postgres::database::DatabaseError;
+use tokio::{signal, try_join};
+
+pub static SNAPSHOT_SINGLETON_ROW_ID: &str = "SINGLETON";
+
+pub struct SnapshotApi {
+    db: Arc<Database>,
+}
+
+#[async_trait]
+impl ReplicatorSnapshot for SnapshotApi {
+    async fn get_snapshot(&self) -> Result<u64, String> {
+        let client = &self.db.get().await.unwrap();
+
+        let sql = r#"SELECT "version" FROM cohort_snapshot WHERE id = $1"#;
+
+        let _prepare = client
+            .prepare_cached(sql)
+            .await
+            .map_err(|e| DatabaseError::prepare(e.to_string(), sql.to_string()).to_string())?;
+        let snapshot_row = client.query_one(sql, &[&SNAPSHOT_SINGLETON_ROW_ID]).await.map_err(|e| e.to_string())?;
+
+        let snapshot: i64 = snapshot_row.get("version");
+
+        Ok(snapshot as u64)
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::builder().format_timestamp_millis().init();
+
+    // 0. Create required items.
+    //  a. Create Kafka consumer
+    let mut kafka_config = KafkaConfig::from_env();
+    kafka_config.group_id = "talos-replicator-dev-1".to_string();
+    let kafka_consumer = KafkaConsumer::new(&kafka_config);
+
+    // b. Subscribe to topic.
+    kafka_consumer.subscribe().await.unwrap();
+
+    // e. Create postgres statemap installer instance.
+    let cfg_db = DatabaseConfig {
+        database: env_var!("COHORT_PG_DATABASE"),
+        host: env_var!("COHORT_PG_HOST"),
+        password: env_var!("COHORT_PG_PASSWORD"),
+        port: env_var!("COHORT_PG_PORT"),
+        user: env_var!("COHORT_PG_USER"),
+        pool_size: env_var_with_defaults!("COHORT_PG_POOL_SIZE", usize, 10),
+    };
+    let database = Database::init_db(cfg_db).await.map_err(|e| e.to_string()).unwrap();
+
+    let pg_statemap_installer = StatemapInstallerImpl {
+        database: Arc::clone(&database),
+        max_retry: env_var_with_defaults!("BANK_STATEMAP_INSTALLER_MAX_RETRY", u32, 3),
+        retry_wait_ms: env_var_with_defaults!("BANK_STATEMAP_INSTALL_RETRY_WAIT_MS", u64, 2),
```

Comment:
the default as 2ms is too small. Should be 2000

Reply:
2ms on retry was installing it successfully on my machine. And as this is just an example, I didn't consider the scenario of deploy + test.  When you test on your machine, if it takes more than 1 attempt, feel free to bump that value. But probably not `2000ms` as that would impact our metrics.

Reply:
2000ms between retry attempts (if I'm reading this right)? Seems a little excessive, no?

Reply:
yeah, I am guessing that was a typo

Reply:
Well, I actually thought this setting is the max wait time for exponential wait-retry.